### DISCUSSION TITLE
Hook up to react native navigation tab?

### DISCUSSION THREAD

**Discussion by talon-i-am** [lukebrandonfarrell/react-native-navigation-drawer-extension#70] [Q&A]
This is more of a general question. Can this be triggered by a react native navigation bottom tab?

**Comment by lukebrandonfarrell** [comment]
@TheTalonHughes I don't see why not, what have you tried so far?

**Comment by talon-i-am** [comment]
@lukebrandonfarrell I am currently using the built in side menu by using Navigation.events().registerBottomTabSelectedListener like so

```
  Navigation.events().registerBottomTabSelectedListener(
    ({ selectedTabIndex }) => {
      if (selectedTabIndex === 3) {
        Navigation.mergeOptions(BottomTabsComponentId, {
          sideMenu: {
            right: {
              visible: true,
            },
          },
        });
      }
    }
  );
  ```

**Comment by lukebrandonfarrell** [comment]
@TheTalonHughes You wouldn't use the side menu view include with RNN, you instead would invoke the RN library,  going to move this to discussions as there isn't any issue here.

**Reply by talon-i-am** [reply]
@lukebrandonfarrell Thanks for moving this to discussion. I can provide some context on what I am trying to achieve if that helps. 

I was initially trying to get a bottom tab to trigger a side menu. I was able to get that somewhat working (trying to figure out how to prevent a tab from navigating and just open a drawer/side menu) but I think I rather use this drawer extension instead. 

As mentioned, I was able to accomplish it with the side menu with the following code

```
  Navigation.events().registerBottomTabSelectedListener(
    ({ selectedTabIndex }) => {
      if (selectedTabIndex === 3) {
        Navigation.mergeOptions(componentId, {
          sideMenu: {
            right: {
              visible: true,
            },
          },
        });
      }
    }
  );
  ```
  
  Sorry for any confusion, I wasn't expecting to use the side menu and this extension together, more-so looking for direction on how to pull off what I am doing with the side menu but with this extension instead. Thanks!
